javaparser-core/src/main/java/com/github/javaparser/ast/body/VariableDeclarator.java

```diff
@@ -85,6 +89,37 @@ public VariableDeclarator(Range range, Type type, SimpleName name, Expression in
         setName(name);
         setInitializer(initializer);
         setType(type);
+        init();
+    }
+
+    private void init() {
+        // We register an observer on the type property. When it is changed the MaximumCommonType is changes as well,
```

Comment:
Well. if you're going to have only one init in the whole project, it could use a much more specific name. Especially if you're going to annotate it with @Initializer or so.